tests/test_outputmanager.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from copy import deepcopy
+
+import pytest
+
+from mreg_cli.outputmanager import remove_dict_key_recursive
+from mreg_cli.types import Json
+
+
+@pytest.mark.parametrize(
+    "inp,key,expected",
+    [
+        pytest.param(
+            {"a": 1, "b": 2},
+            "a",
+            {"b": 2},
+            id="simple_single_key_dict",
+        ),
+        pytest.param(
+            {"a": {"b": 2, "c": 3}, "d": 4},
+            "b",
+            {"a": {"c": 3}, "d": 4},
+            id="nested_dict_removal",
+        ),
+        pytest.param(
+            [{"a": 1}, {"a": 2}, {"b": 3}],
+            "a",
+            [{}, {}, {"b": 3}],
+            id="list_of_dicts",
+        ),
+        pytest.param(
+            {"a": [{"b": 1, "c": 2}, {"b": 3, "d": 4}], "e": {"b": 5, "f": {"b": 6}}},
+            "b",
+            {"a": [{"c": 2}, {"d": 4}], "e": {"f": {}}},
+            id="complex_nested_structure",
+        ),
+        pytest.param(
+            {"a": 1, "b": 2},
+            "c",
+            {"a": 1, "b": 2},
+            id="key_not_present",
+        ),
+        pytest.param(
+            {},
+            "a",
+            {},
+            id="empty_dict",
+        ),
+        pytest.param(
+            [],
+            "a",
+            [],
+            id="empty_list",
+        ),
+        pytest.param(
+            {"a": 1, "b": "string", "c": True, "d": None, "e": 1.5, "f": {"a": 2}},
+            "a",
+            {"b": "string", "c": True, "d": None, "e": 1.5, "f": {}},
+            id="mixed_types_with_nested_removal",
+        ),
+        pytest.param(
+            [[[{"a": 1}]], [[{"a": 2}]]],
+            "a",
+            [[[{}]], [[{}]]],
+            id="deeply_nested_lists",
+        ),
+    ],  # type: ignore
+)
+def test_remove_dict_key_recursive(inp: Json, key: str, expected: Json) -> None:
+    """Test remove_dict_key_recursive with a variety of inputs."""
+    # Call the function
```

Comment:
Of all the things LLMs decide to comment, stuff like this and the assert below is by far the most "huh" in my book.

Reply:
Agreed, will remove.